### ISSUE TITLE
Query: Oracle: Indexof_with_emptystring fails

### ISSUE THREAD

**Issue by smitpatel** [dotnet/efcore#10563]
empty string creates parameter and it is sent to server. When it returns it comes as null. (Oracle doesn't differentiate between null/empty string) and we use the value read from server in IndexOf function on client. Which throws Value cannot be null
```
Test 'Microsoft.EntityFrameworkCore.Query.SimpleQueryOracleTest.Indexof_with_emptystring' failed: System.ArgumentNullException : Value cannot be null.
Parameter name: value
	at System.String.IndexOf(String value, Int32 startIndex, Int32 count, StringComparison comparisonType)
	at lambda_method(Closure , QueryContext , ValueBuffer )
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Relational\Query\ExpressionVisitors\Internal\ProjectionShaper.cs(97,0): at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.ProjectionShaper.TypedProjectionShaper`3.Shape(QueryContext queryContext, ValueBuffer valueBuffer)
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Relational\Query\Internal\QueryingEnumerable.cs(129,0): at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.BufferlessMoveNext(DbContext _, Boolean buffer)
	D:\code\SecondEF\EntityFrameworkCore\samples\OracleProvider\src\OracleProvider\Storage\Internal\OracleExecutionStrategy.cs(39,0): at Microsoft.EntityFrameworkCore.Storage.Internal.OracleExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Relational\Query\Internal\QueryingEnumerable.cs(81,0): at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore\Query\Internal\LinqOperatorProvider.cs(141,0): at Microsoft.EntityFrameworkCore.Query.Internal.LinqOperatorProvider.ExceptionInterceptor`1.EnumeratorExceptionInterceptor.MoveNext()
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Specification.Tests\TestUtilities\QueryAsserter.cs(70,0): at Microsoft.EntityFrameworkCore.TestUtilities.QueryAsserter`1.AssertSingleResult[TItem1](Func`2 actualQuery, Func`2 expectedQuery, Action`2 asserter, Int32 entryCount)
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Specification.Tests\Query\QueryTestBase.cs(34,0): at Microsoft.EntityFrameworkCore.Query.QueryTestBase`1.AssertSingleResult[TItem1](Func`2 actualQuery, Func`2 expectedQuery, Action`2 asserter, Int32 entryCount)
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Specification.Tests\Query\QueryTestBase.cs(26,0): at Microsoft.EntityFrameworkCore.Query.QueryTestBase`1.AssertSingleResult[TItem1](Func`2 query, Action`2 asserter, Int32 entryCount)
	D:\code\SecondEF\EntityFrameworkCore\src\EFCore.Specification.Tests\Query\SimpleQueryTestBase.Functions.cs(839,0): at Microsoft.EntityFrameworkCore.Query.SimpleQueryTestBase`1.Indexof_with_emptystring()
	D:\code\SecondEF\EntityFrameworkCore\samples\OracleProvider\test\OracleProvider.FunctionalTests\Query\SimpleQueryOracleTest.cs(48,0): at Microsoft.EntityFrameworkCore.Query.SimpleQueryOracleTest.Indexof_with_emptystring()
```

**Comment by ajcvickers** [comment]
Triage: Assigning to @anpete to understand the root cause of the issue and:
* If it is an Oracle issue, then:
  * Either fix in the sample
  * Disable the tests and leave as work to do for somebody porting the provider to production code
* If it is an issue in EF Core, then:
  * Fix the issue in EF Core, or re-riage

**Comment by ajcvickers** [comment]
This is an issue identified on the Oracle sample provider. We don't plan to fix this in the sample, but we would be happy for a community contributor to fix it. Also, anyone taking the sample as a starting point for production code should consider fixing this issue.

**Comment by ralmsdeveloper** [comment]
I think this can be closed, fixed in #10579

/cc @ajcvickers

**Comment by ajcvickers** [comment]
Closing old issue as this is no longer something we intend to implement.
